.github/workflows/release.yml

```diff
@@ -47,3 +47,18 @@ jobs:
                 target: ${{ matrix.target }}
                 profile: release-lto
                 token: ${{ secrets.GITHUB_TOKEN }}
+
+    publish-crates:
+        name: Publish crates to crates.io
+        needs: upload-assets
+        runs-on: ubuntu-latest
+        steps:
+            - name: Extract version from tag
+              id: version
+              run: |
+                VERSION=${GITHUB_REF#refs/tags/v}
+                echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+            - uses: ./.github/workflows/publish-crates.yml
+              with:
+                  version: ${{ steps.version.outputs.version }}
```

Comment:
Once we publish these crates we can't really update them, right? If so, then this "publish crates" job should be executed only when we publish the actual release on GitHub?

Otherwise it's possible that:

- we tag a new release in git
- this workflow creates the draft release and builds binary assets, then goes on to publish the crates
- the Docker build however fails and we won't be able to fix it _properly_ because these crates can't be re-published

So I this step should only be done after we've actually published the release in a separate workflow triggered by a new release being published:

```yaml
on:
  release:
    types: [published]
```

Reply:
Tracking this in a new issue: #2509